### ISSUE TITLE
Arch Linux steam stucks at "Installing breakpad exception handler"

### ISSUE THREAD

**Issue by ghost** [ValveSoftware/steam-for-linux#3954]
I'm on Arch 64bit with an NVidia Geforce GTX 660 and with nvidia-libgl nvidia-utils lib32-nvidia-libgl lib32-nvidia-utils drivers.

I'm trying to start steam:

```
steam
/home/tekkkz/.local/share/Steam/steam.sh: line 161: VERSION_ID: unbound variable
/home/tekkkz/.local/share/Steam/steam.sh: line 161: VERSION_ID: unbound variable
Running Steam on arch  64-bit
/home/tekkkz/.local/share/Steam/steam.sh: line 161: VERSION_ID: unbound variable
STEAM_RUNTIME is enabled automatically
Installing breakpad exception handler for appid(steam)/version(0)
```

And there nothing happens.
How to solve this?

**Comment by gdrewb-valve** [comment]
This looks like a duplicate of #3951, see if that helps.

**Comment by Hazerdous** [comment]
Running into the same problem.
Archlinux 64bit, Nvidia GeForce GTX 970. Beta versions of the Nvidia proprietary drivers (nvidia-beta, nvidia-libgl-beta, lib32-nvidia-libgl-beta, etc) from the AUR.
Downgrading to an earlier version of the drivers (as suggested by the referenced issue) is impossible, as any versions earlier than the beta (v358) do not work with my setup

The problem appears to be a missing client version, normally the 0 in `appid(steam)/version(0)` is a client version number, such as `1352224866_client`

I can't find where the `Installing breakpad exception handler...` echo is from, I checked the steam.sh and the /bin/steam files.

**Comment by Tekkkz** [comment]
Please give me the whole cmdline output when you run steam. I solved the issue and can give you help if you give me all information.

Mit Outlook<http://taps.io/outlookmobile> für Android gesendet



On Thu, Nov 12, 2015 at 6:48 PM -0800, "Daniel" <notifications@github.com<mailto:notifications@github.com>> wrote:


Running into the same problem.
Archlinux 64bit, Nvidia GeForce GTX 970. Beta versions of the Nvidia proprietary drivers (nvidia-beta, nvidia-libgl-beta, lib32-nvidia-libgl-beta, etc) from the AUR.
Downgrading to an earlier version of the drivers (as suggested by the referenced issue) is impossible, as any versions earlier than the beta (v358) do not work with my setup

The problem appears to be a missing client version, normally the 0 in appid(steam)/version(0) is a client version number, such as 1352224866_client

I can't find where the Installing breakpad exception handler... echo is from, I checked the steam.sh and the /bin/steam files.

-

**Comment by Hazerdous** [comment]
Running Steam on arch 4.2.5-1-ARCH 64-bit
    STEAM_RUNTIME is enabled automatically
    Installing breakpad exception handler for appid(steam)/version(0)
There were the errors about invalid variables too, but I fixed those by adding a failsafe to steam.sh and adding `VERSION_ID="$(uname -r)"` to /etc/os-release

**Comment by Tekkkz** [comment]
Man. Type "steam" in terminal and give me the WHOLE OUTPUT.

Mit Outlook<http://taps.io/outlookmobile> für Android gesendet



On Thu, Nov 12, 2015 at 10:05 PM -0800, "Daniel" <notifications@github.com<mailto:notifications@github.com>> wrote:


Running Steam on arch 4.2.5-1-ARCH 64-bit STEAM_RUNTIME is enabled automatically Installing breakpad exception handler for appid(steam)/version(0)

There were the errors about invalid variables too, but I fixed those by adding a failsafe to steam.sh and adding VERSION_ID="$(uname -r)" to /etc/os-release

-

**Comment by Hazerdous** [comment]
http://i.imgur.com/OtitAZ6.png The entire output, it gets stuck there

**Comment by Tekkkz** [comment]
Clear your local steam files, dont modify anything and try again.

Mit Outlook<http://taps.io/outlookmobile> für Android gesendet



On Thu, Nov 12, 2015 at 10:33 PM -0800, "Daniel" <notifications@github.com<mailto:notifications@github.com>> wrote:


http://i.imgur.com/OtitAZ6.png The entire output, it gets stuck there

-

**Comment by tocsinDE** [comment]
Got the exact same error messages like @Hazerdous. Running Steam on arch 4.2.5-1-ARCH 64-bit, GeForce GTX 950M, stable drivers.

**Comment by Hazerdous** [comment]
[daniel@Daniel-PC ~] $ rm -r .steam .local/share/Steam/
    [daniel@Daniel-PC ~] $ steam
    Setting up Steam content in /home/daniel/.local/share/Steam
    /home/daniel/.local/share/Steam/steam.sh: line 102: no match: **:
    Running Steam on arch 4.2.5-1-ARCH 64-bit
    STEAM_RUNTIME is enabled automatically
    Installing breakpad exception handler for appid(steam)/version(0)

**Comment by Hazerdous** [comment]
I found something interesting. When attempting to run steam when no X session is found (i.e, via ssh) the process shows a bit more output.

    Running Steam on arch 4.2.5-1-ARCH 64-bit
    STEAM_RUNTIME is enabled automatically
    Installing breakpad exception handler for appid(steam)/version(0)
    Assert( Assertion Failed: Could not open connection to X ):/home/buildbot/buildslave_steam/steam_rel_client_ubuntu12_linux/build/src/steamexe/./updateui_xwin.cpp:337
    Installing breakpad exception handler for appid(steam)/version(1.0)
    sh: /home/daniel/.local/share/Steam/steam_msg.sh: No such file or directory
    crash_20151122004213_3.dmp[10124]: Uploading dump (out-of-process)
    /tmp/dumps/crash_20151122004213_3.dmp
    crash_20151122004213_3.dmp[10124]: Finished uploading minidump (out-of-process): success = no
    crash_20151122004213_3.dmp[10124]: error: libcurl.so: cannot open shared object file: No such file or directory
    crash_20151122004213_3.dmp[10124]: file ''/tmp/dumps/crash_20151122004213_3.dmp'', upload no: ''libcurl.so: cannot open shared object file: No such file or directory''
    ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
    [2015-11-22 00:42:13] Startup - updater built May  5 2015 12:15:27
    /home/buildbot/buildslave_steam/steam_rel_client_ubuntu12_linux/build/src/steamexe/./updateui_xwin.cpp (337) : Assertion Failed: Could not open connection to X
    /home/buildbot/buildslave_steam/steam_rel_client_ubuntu12_linux/build/src/steamexe/./main.cpp (513) : Assertion Failed: failed to initialize update status ui, or create initial window

**Comment by tocsinDE** [comment]
While looking through the [wiki entry about Bumblebee](https://wiki.archlinux.org/index.php/Bumblebee#Installing_Bumblebee_with_Intel.2FNVIDIA) I figured out I installed the wrong 32bit graphics driver. Installing lib32-mesa-libgl and uninstalling lib32-nvidia-libgl fixed it perfectly.

**Comment by Oxalin** [comment]
@Tekkkz Are you still experiencing this bug? Could you try to see if there are any missing libraries (using ldd on the steam executable)? What happens if you try to override Steam's runtime libraries? Was your system installed with the mesa libraries (@tocsinDE, you may be concerned here) and updated to the NVidia librairies (look at this particular section https://wiki.archlinux.org/index.php/Steam#You_are_missing_the_following_32-bit_libraries.2C_and_Steam_may_not_run:_libGL.so.1)?

You can find answers to all these questions by looking at the page: https://wiki.archlinux.org/index.php/Steam

**Comment by tafia** [comment]
I'm on debian stretch/intel nuc and I have a similar error:

```
/home/johann/.local/share/Steam/steam.sh: line 161: VERSION_ID: unbound variable
/home/johann/.local/share/Steam/steam.sh: line 161: VERSION_ID: unbound variable
Running Steam on debian  64-bit
/home/johann/.local/share/Steam/steam.sh: line 161: VERSION_ID: unbound variable
STEAM_RUNTIME is enabled automatically
Installing breakpad exception handler for appid(steam)/version(0)
*** Error in `/home/johann/.local/share/Steam/ubuntu12_32/steam': free(): invalid pointer: 0xf775c000 ***
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[2015-12-29 16:11:40] Startup - updater built Nov 17 2015 12:33:46
assert_20151229161140_1.dmp[2501]: Uploading dump (out-of-process)
/tmp/dumps/assert_20151229161140_1.dmp
/home/johann/.local/share/Steam/steam.sh: line 756:  2498 Aborted                 $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
assert_20151229161140_1.dmp[2501]: Finished uploading minidump (out-of-process): success = no
assert_20151229161140_1.dmp[2501]: error: libcurl.so: cannot open shared object file: No such file or directory
assert_20151229161140_1.dmp[2501]: file ''/tmp/dumps/assert_20151229161140_1.dmp'', upload no: ''libcurl.so: cannot open shared object file: No such file or directory''
```

I've tried installing both from the .deb file and from debian repos.

**Comment by Oxalin** [comment]
@Hazerdous and @tafia , both of you have an error about a missing library:
crash_20151122004213_3.dmp[10124]: file ''/tmp/dumps/crash_20151122004213_3.dmp'', upload no: ''libcurl.so: cannot open shared object file: No such file or directory''
and 
assert_20151229161140_1.dmp[2501]: error: libcurl.so: cannot open shared object file: No such file or directory

Is libcurl installed (32 bit)?

**Comment by tafia** [comment]
@Oxalin, 32 bit version wasn't installed.
I now have another message:
```
/home/johann/.local/share/Steam/steam.sh: line 161: VERSION_ID: unbound variable
/home/johann/.local/share/Steam/steam.sh: line 161: VERSION_ID: unbound variable
Running Steam on debian  64-bit
/home/johann/.local/share/Steam/steam.sh: line 161: VERSION_ID: unbound variable
STEAM_RUNTIME is enabled automatically
Installing breakpad exception handler for appid(steam)/version(0)
*** Error in `/home/johann/.local/share/Steam/ubuntu12_32/steam': free(): invalid pointer: 0xf77be000 ***
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[2015-12-30 08:48:28] Startup - updater built Nov 17 2015 12:33:46
assert_20151230084828_1.dmp[8748]: Uploading dump (out-of-process)
/tmp/dumps/assert_20151230084828_1.dmp
/home/johann/.local/share/Steam/steam.sh: line 756:  8745 Aborted                 $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
assert_20151230084828_1.dmp[8748]: Finished uploading minidump (out-of-process): success = yes
assert_20151230084828_1.dmp[8748]: response: CrashID=bp-79161069-08f2-4ea1-bdca-cc5812151229
assert_20151230084828_1.dmp[8748]: file ''/tmp/dumps/assert_20151230084828_1.dmp'', upload yes: ''CrashID=bp-79161069-08f2-4ea1-bdca-cc5812151229''
```

**Comment by Oxalin** [comment]
@tafia could you have a look at this link to remove the first error about VERSION_ID being undefined:
https://wiki.archlinux.org/index.php/Steam#VERSION_ID:_unbound_variable

Then, could you run the proposed commands here to see if there are still missing libraries: https://wiki.archlinux.org/index.php/Steam#Using_native_runtime

**Comment by tafia** [comment]
Sorry, didn't see your answer.
It looks to be working for the VERSION _ID but not using native libraries:
```
Running Steam on debian 2016-01-03 64-bit
STEAM_RUNTIME is enabled automatically
Installing breakpad exception handler for appid(steam)/version(0)
*** Error in `/home/johann/.local/share/Steam/ubuntu12_32/steam': free(): invalid pointer: 0xf77a1000 ***
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[2016-01-04 13:22:47] Startup - updater built Nov 17 2015 12:33:46
assert_20160104132247_1.dmp[5505]: Uploading dump (out-of-process)
/tmp/dumps/assert_20160104132247_1.dmp
/home/johann/.local/share/Steam/steam.sh: line 756:  5502 Aborted                 $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
assert_20160104132247_1.dmp[5505]: Finished uploading minidump (out-of-process): success = yes
assert_20160104132247_1.dmp[5505]: response: CrashID=bp-e05a65e1-4e1d-4ede-85fa-60ecf2160103
assert_20160104132247_1.dmp[5505]: file ''/tmp/dumps/assert_20160104132247_1.dmp'', upload yes: ''CrashID=bp-e05a65e1-4e1d-4ede-85fa-60ecf2160103''
```

**Comment by Oxalin** [comment]
@tafia Since you are not using the native libraries, you are missing a dependency OR there is a conflict with some libraries your system is providing.

1- If I understand correctly what you wrote previously, you seem to be using an Intel GPU. Thus, you are relying on mesa drivers. Could you try what's under: https://wiki.archlinux.org/index.php/Steam#Steam_runtime_issues

You could begin with the 3rd command (LD_PRELOAD=...) where you'll probably have to replace "$LIB" by "i386-linux-gnu" (please verify first where your 32 bit library folder is, I'm not used to debian multiarch folders anymore, it could be different).

2- Another thing you can try, just in case, could you make sure en_US locales are available:
https://forum.manjaro.org/index.php?topic=19259.msg175257#msg175257

**Comment by HanKruiger** [comment]
I had the same problem. However, since I am using Bumblebee, Steam was unable to see my NVIDIA libraries. Running Steam with `optirun steam` fixed the problem for me. Hope this will help someone.

**Comment by redlotus** [comment]
@HanKruiger  It works for me too, thanks.

**Comment by johan-bjareholt** [comment]
I do not have bumblebee but still have this issue. Help?

**Comment by Oxalin** [comment]
@johan-bjareholt , could give us more details about what you tried, your setup, if you are using STEAM_RUNTIME or disabling it, etc.?

**Comment by johan-bjareholt** [comment]
I actually solved it yesterday, it was fully my mistake.

Hadn't rebooted my computer in 3 weeks and have had multiple nvidia driver
updates since so the issue was that it issues with libGL.so so I couldn't
even start glxgears. Everything worked fine after a reboot.

Strange that steam just hangs in this case instead of showing a error
dialog or something though.

**Comment by FeralBytes** [comment]
I realized that this is old, but it is still a problem. Steam hangs when using Nvidia Optimus and I have my system currently using Intel graphics; but have requested the system to switch to Nvidia graphics; but have not yet logged out and back in to complete the switch. It may actually happen if I am just on Intel graphics but maybe this is the specific edge case. Either way it would be good if Steam would at least show a dialogue and state that it was unable to get a drawable surface from the GPU.

**Comment by MasterLampe** [comment]
Hey guys,
I'm also experiencing this weird problem! So far I've been unable to solve it. My steam client ran totally smooth all the time, but suddenly from one day to the other, without even doing an update in between or something, it just stopped working. I tried a fresh reinstall of steam, I tried different nvidia drivers, I tried what was said in the arch wiki article about having problems with the runtime, but no matter I keep getting this: 
`Running Steam on arch rolling 64-bit
STEAM_RUNTIME is disabled by the user
Installing breakpad exception handler for appid(steam)/version(0)
`
I don't even know if it's a GPU driver problem because the runtime doesn't give me any errors or so about the problem. It just keeps hanging there. Just to let you know: I have a GTX 1060 and I have nvidia proprietary drivers installed (current version 381.22-4) and also lib32-nvidia-utils. I also have multilib and multilib-test enabled in pacman.conf.

I even tried to check through the steam.sh but I always get confused when reading bash xD.

I really would appreciate some help here!

Cheers

**Comment by archenroot** [comment]
I am on Gentoo system, having just 2 Nvidia 1080Ti and 2 Xeons (so no integrated GPU) -> No optimus case.

In my case this issue was caused because I updated nvidia drivers and without reboot  tried to run steam. Simple reboot fixed the issue.

**Comment by MasterLampe** [comment]
I also came across why my steam did not launch a few months ago and later started working properly again: 

If you use a Graphics card from nvidia make sure you have the nvidia driver AND the 32 bit driver installed (for example nvidia, nvidia-utils and lib32-nvidia-utils). Also they MUST have the same version number. 

In my case, I had the lib32 package installed from the multilib-testing repo, which was updated faster than my actual "nvidia"-package, so they had different versions and steam would hang at "installing breakpoint exception handler". So I downgraded the lib32 package from my pacman cache and everything worked fine. So if you update these packages, always do so simultaneously, so that they have the same version number.
